### PR TITLE
E2 sound extensions

### PR DESCRIPTION
**Pull request by stepa2**
- Exports some SoundCore functions
- Adds SoundCore functions from modified xtrasound core (https://steamcommunity.com/sharedfiles/filedetails/?id=2221932128)
- Adds EmitSound core - a simplier way to play sounds

### CODE REVIEW

**Comment by thegrb93** [comment]
I don't think we need a new extension for the emitsound functions. Thoughts, @Vurv78 ?

**Comment by Vurv78** [comment]
I agree, don't see any reason for them to be separate, especially if having them separate is will need you to export fields in `E2Lib`.

**Comment by Vurv78** [comment]
As for the rest of the unused exported fields, why? Maybe I'd understand IsAllowed in case you want to configure that as an admin, but why not just make a hook then? And I *really* don't like the idea of other addons depending on internal code like that (especially with #2555)

**Comment by stepa2** [comment]
Originally soundcore stuff was exported because extra sound functions (`soundDsp`, etc.) were in separate core.

File: lua/wire/client/e2descriptions.lua [hunk 1]
```diff
@@ -1525,6 +1525,8 @@ E2Helper.Descriptions["soundVolume(s)"] = "Returns the volume of the sound at th
 E2Helper.Descriptions["soundPlaying(n)"] = "Returns 1 if the sound at the index is playing, 0 if not"
 E2Helper.Descriptions["soundPlaying(s)"] = "Returns 1 if the sound at the index is playing, 0 if not"
 
+E2Helper.Descriptions["emitSound"] = "Plays sound on entity. Note that one file can only be played once in a time."
```
Comment (Vurv78) [inline lua/wire/client/e2descriptions.lua:1528]: These are invalid, need `emitSound(e:snnn)`, `emitSound(e:snn)`, `emitSound(e:sn)`, `emitSound(e:s)`, `emitSoundStop(e:s)`

**Review by Vurv78 — CHANGES_REQUESTED** [review]

**Pull request merged**